### PR TITLE
Unity serialisation changes

### PR DESCRIPTION
**Pull request by fractalwrench**
## Goal

Simplify bridging of metadata for Unity and other notifiers.

## Changeset

- Update serialisation unit tests (removed setter tests as this would just be testing `java.util.Map`)
- Add `mapToJson` utility for constructing JSON in tests
- Combine `AppData` + `AppDataSummary` into one class
- Combine `DeviceData` + DeviceDataSummary` into one class
- Remove public getters/setters/fields on above classes, and return a `Map` instead

Reduce the number of accessors on `Client` to:

```
AppData getAppData();
DeviceData getDeviceData();
```

Add the following accessors on AppData/DeviceData:

```
Map<String, Object> getAppData()
Map<String, Object> getAppDataSummary()
Map<String, Object> getDeviceData()
Map<String, Object> getDeviceDataSummary()
```

- Update `Error` and `SessionTrackingPayload` to serialise a map
- Update `NativeInterface`, `Client`, other misc classes to access information from map
- Combined SessionTrackingPayload constructors to reduce method count

Existing JSON serialisation code in `MetaData` has been refactored into a separate class called `ObjectJsonStreamer`. This serialises the `Map` fields in the Error and Session payloads.

## Tests

Ran unit tests and mazerunner scenario. Inspected error payload manually in dashboard.

## Review

For the submitter, initial self-review:

- [ ] Commented on code changes inline explain the reasoning behind the approach
- [ ] Reviewed the test cases added for completeness and possible points for discussion
- [ ] A changelog entry was added for the goal of this pull request
- [ ] Check the scope of the changeset - is everything in the diff required for the pull request?
- This pull request is ready for:
  - [ ] Initial review of the intended approach, not yet feature complete
  - [ ] Structural review of the classes, functions, and properties modified
  - [ ] Final review

For the pull request reviewer(s), this changeset has been reviewed for:

- [x] Consistency across platforms for structures or concepts added or modified
- [x] Consistency between the changeset and the goal stated above
- [x] Internal consistency with the rest of the library - is there any overlap between existing interfaces and any which have been added?
- [x] Usage friction - is the proposed change in usage cumbersome or complicated?
- [x] Performance and complexity - are there any cases of unexpected O(n^3) when iterating, recursing, flat mapping, etc?
- [x] Concurrency concerns - if components are accessed asynchronously, what issues will arise
- [x] Thoroughness of added tests and any missing edge cases
- [x] Idiomatic use of the language

### CODE REVIEW

File: sdk/src/androidTest/java/com/bugsnag/android/ClientTest.java [hunk 1]
```diff
@@ -303,28 +303,17 @@ public void testBreadcrumbStoreNotModified() {
...
-        assertNotNull(appDataSummary);
-        assertNotEquals(appData.getAppDataSummary(), appData);
-    }
-
     @Test
     public void testAppDataMetaData() {
         Client client = generateClient();
         Map<String, Object> app = client.getAppData().getAppDataMetaData();
-        assertEquals(5, app.size());
+        assertEquals(6, app.size());
```
Comment (kattrali) [inline sdk/src/androidTest/java/com/bugsnag/android/ClientTest.java]: What is the 6th item? Should there be an assertion for that?

Comment (fractalwrench) [reply sdk/src/androidTest/java/com/bugsnag/android/ClientTest.java]: The `codeBundleId` (which is null by default). I can add an assertion for this.

Comment (fractalwrench) [reply sdk/src/androidTest/java/com/bugsnag/android/ClientTest.java]: I was thinking of the wrong object - it's actually `activeScreen`, which will be null in the unit test. I've added a check to ensure the key is present in the map.
